src/thor/matrix_action.cc

```diff
@@ -66,15 +66,21 @@ std::string thor_worker_t::matrix(Api& request) {
   // lambdas to do the real work
   std::vector<TimeDistance> time_distances;
   auto costmatrix = [&]() {
-    return costmatrix_.SourceToTarget(options.sources(), options.targets(), *reader, mode_costing,
-                                      mode, max_matrix_distance.find(costing)->second);
+    time_distances =
+        costmatrix_.SourceToTarget(options.sources(), options.targets(), *reader, mode_costing, mode,
+                                   max_matrix_distance.find(costing)->second);
+    costmatrix_.clear();
```

Comment:
clear the algos

Reply:
we want to do these over in worker.cc in the clear() method that way it happens not in the request path and rather only after returning the response to the requestor

Reply:
done